fetch-client/types.ts

```diff
@@ -44,7 +44,8 @@ export interface IResponse {
  * options for building a signed request.
  */
 export interface ISignedRequestOptions {
-  signer?: ISigner
+  signer?: ISigner,
+  headers?: Record<string, string>
```

Comment:
in general any given http header name may appear more than one time. i.e. there can be many headers all named `Cookie`.

This is why if you look elsewhere in this file I tend to type it as `Iterable<[name:string,value:string]>` which accomodates an array of 2-tuples representing each header. Unlike a hash map it allows multiple headers with same `name`.

later I will probably extend the type to also support what I used e.g. in `IResponse['headers']`
This still is a good thing to support as syntax sugar though.